bagel/pheno_utils.py

```diff
@@ -35,6 +39,39 @@ def validate_portal_uri(portal: str) -> Optional[str]:
     return portal
 
 
+def load_pheno(input_p: Path) -> pd.DataFrame | None:
+    """Load a .tsv pheno file and do some basic validation."""
+    if input_p.suffix == ".tsv":
+        pheno_df: DataFrame = pd.read_csv(
+            input_p, sep="\t", keep_default_na=False, dtype=str
+        )
+
+        if pheno_df.shape[1] > 1:
+            return pheno_df
+        elif len(pheno_df.columns[0].split(",")) > 1:
+            # We only extracted one column, this might mean that we have a sneaky .csv
+            raise ValueError(
+                f"Your phenotypic input file {input_p} has only one column "
+                f"and is therefore not valid as a neurobagel phenotypic file. "
+                f"Note that your phenotypic input file also looks like a .csv file "
+                f"as it contains several ',' commas. It is possible that "
+                f"you have accidentally renamed a .csv file as a .tsv"
+                " Please provide a valid .tsv pheno file!"
+            )
+        else:
+            warnings.warn(
+                f"Your phenotypic input file {input_p} has only one column."
+                " Such a file is not a valid Neurobagel phenotypic data file."
+                " Please check your input again!"
+            )
+            return pheno_df
```

Comment:
🍒 Just curious - any reason you chose to raise a warning here instead of just erroring out and exiting the program?

I don't mind letting single-column TSVs still go through the CLI if there is nothing else apparently wrong with the contents, but if this is the intended behaviour we may want to change the warning phrasing to be different than the actually _invalid_ input scenario above (which _does_ error out), e.g.: 

```suggestion
                f"Your phenotypic input file {input_p} has only one column."
                " We recommend double checking your input."
            )
            return pheno_df
```

Reply:
No I think you're right. Let's error out. 

Speaking of that, I noticed you used this syntax recently for erroring: 

https://github.com/neurobagel/bagel-cli/blob/4434745de050c75157635d81fe877e5c3531b471/bagel/utility.py#L16-L21

Do you think we should adopt that across the board? What's the benefit?

Reply:
@surchs aha, I adopted that syntax because I was looking at this docs [page](https://typer.tiangolo.com/tutorial/terminating/) and thought our scenario of "file already exists" was similar to the example they provided and so wanted to try it. 😛

I think the benefit is that the output message looks really clean, and there's no traceback or anything like that which makes sense for an upfront sanity check like this. On the other hand, I think this kind of termination is probably not the most suitable for handling more complex input errors - the default exit code is still 0, so it's not treated like other exceptions like `ValueError` etc unless we manually pass in error codes. This might make testing a little trickier as well: from my brief playing around with it for this PR, our normal syntax of using `with pytest.raises` doesn't seem to capture these exit exceptions (?).

All in all, I think most of the custom errors (TSV vs JSON incompatibility, missing keys, etc.) we catch probably still are conceptually better handled as normal exceptions, but we can revisit this if needed as part of https://github.com/neurobagel/bagel-cli/issues/218.

Reply:
Let's leave this for #218